### DISCUSSION TITLE
Solved: Closed doors and elevators

### DISCUSSION THREAD

**Discussion by BernardoFPereira** [scandum/tintin#120] [Script help]
Hey! New user here. Awesome work on TinTin!

Gotta admit, it was a bit overwhelming at first, mostly because I'm new to MUDs as well. But it's grown on me quite a bit.
Recently started mapping a MUD and ran into a couple situations.
Apologies if any of the questions are a bit too basic.

### **1st-** Saw this on another post.
> You can set data on rooms and exits, next code a loop to run through each room/exit to set the avoid flag.
> 
> Commands of interest:
> 
> ```
> #map list {variable} {varname}
> #map set roomdata {simple or nested variable}
> #map set roomdata {varname}
> #map exit <dir> set {simple or nested variable}
> #map exit <dir> get {varname}
> ```
> 
> Also an option to store the data in a table. Let me know if you need a more practical example.

Would something with those work to prevent the map to move through closed doors? I thought maybe having a trigger to `#map undo` if I get any of the "can't go there" messages from the game. Would something like that be viable at all? If so, how would one go about it?

### **2nd-** The MUD in question has _elevators_.
This is a compound problem, starting with the fact that the elevator doors are closed automatically and are opened by a `push button` command. On that front, if the above would work, or some way to detect open/closed door is available, I'd imagine I could set up something for detection of 'elevator's here, doors opened' messages to trigger opening or closing the way for the map itself. Doesn't seem like much of a hurdle. Does any of this make sense? I'm thinking through it as I type.

Then the real fun begins, the moving about floors. If that problem was faced before I'd love tips. Luckily, the game also have a `push <n>` to press a button to a corresponding floor and go there. So about that my idea so far is, use `#map dig u/d new` and hook something up to detect the `push <n>` commands and trigger `#map goto <target vnum>`. It feels a bit cumbersome and that means I'd probably have to hard code all vnums, wich isn't that big of a deal so far, but I have no idea how to properly make TinTin scripts or if any of this is viable at all.

**Comment by scandum** [comment]
Making a trigger to `#map undo` is indeed a good option.

As for the elevator, you can make an exit named `push 1` and have it link to the elevator destination. The name of an exit is the command it triggers on, and the command of an exit are the commands executed, which is often the same as the name.

**Reply by scandum** [reply]
You kind of can. You can set a delay on an exit with `#map exit` but it is only honored if you use #map run with a default delay. Something like `#map run {some place} 0.25` should work for most muds that process one command every 0.25 seconds.

**Reply by BernardoFPereira** [reply]
Sorry, swapped the reply to a new comment.

Ah cool. Not sure that fits the purpose here. It's not really necessary, too.
The MUD I'm mapping has a small delay to move you between floors when on an elevator, was hoping for a way to match that.

**Comment by BernardoFPereira** [comment]
Awesome! Thank you immensely for the timely response. I think the elevator movement will be solved with this.
Just have to iron out out how to catch the game output now, then I believe placing an `#action` to check if the output matches any the "move denied" messages and trigger `#map undo` if they match shouldn't be too hard.

**Comment by BernardoFPereira** [comment]
It worked beautifully! Can't thank you enough.
Follow-up question. Can I add a delay to the passage between specific doors?
